authenticate/authenticate.go

```diff
@@ -135,13 +148,13 @@ func New(opts config.Options) (*Authenticate, error) {
 		cookieSecret:     decodedCookieSecret,
 		cookieCipher:     cookieCipher,
 		cookieOptions:    cookieOptions,
-		sessionStore:     cookieStore,
+		sessionStore:     cacheStore,
 		encryptedEncoder: encryptedEncoder,
-		sessionLoaders:   []sessions.SessionLoader{qpStore, headerStore, cookieStore},
-		sessionStores:    []sessions.SessionStore{cookieStore, qpStore},
+		sessionLoaders:   []sessions.SessionLoader{cacheStore, qpStore, headerStore, cookieStore},
 		// IdP
 		provider: provider,
 
 		templates: template.Must(frontend.NewTemplates()),
+		addr:      opts.Addr,
```

Comment:
Will this work as intended with defaults for the call to groupcache.NewHTTPPoolOpts() in handlers?

Reply:
In this PR `Addr`, and `groupcache.NewHTTPPoolOpts` are basically nops AFAIK as the pool is never used since we only have local cache.  We would need to support one of the following to be able to have more than 1 instance of the authenticate service. 

1) static peer loading via a configuration setting
2) dynamic peer loading via something like gossip
3) both?

I had planned to look at gossip protocol (todo), but it feels a bit wrong to break N-authenticate service configs on master. Thoughts?

Reply:
I've removed references to the pool since we aren't using it.

Reply:
👍